src/stores/code.ts

```diff
@@ -24,6 +26,63 @@ export const codeStore = defineStore({
     setMyCode(value: string): void {
       this.writeCode = value;
     },
+    setNextIndexCode(i: number): void {
+      this.correctcode = this.fullcode.substring(0, i);
```

Comment:
```suggestion
      this.setCorrectCode(this.getFullCodeItoStart(i));
```
この処理が上の２つのメソッドを使うことでこのように、変更できると思いました。
gettersはキャッシュして処理してくれるらしいので、ある値から別の値を抽出、検索する処理は、ここで定義した方がよいと思いました。
ただ書き換えてうまく動作しない場合があればこのままでよいと思います。